src/ui/editortabwidget.cpp

```diff
@@ -39,6 +40,18 @@ int EditorTabWidget::addEditorTab(bool setFocus, const QString &title)
     return this->rawAddEditorTab(setFocus, title, 0, 0);
 }
 
+bool EditorTabWidget::eventFilter(QObject* object, QEvent* ev)
+{
+    if(object == tabBar() && ev->type() == QEvent::FocusIn) {
```

Comment:
Status bar updates weren't operating properly... so it seemed necessary considering the JS was failing to be called because of it.... I'll try to be a little less heavy-handed with my code in the future...  I'm highly aware that most of my PRs are extensive in scope, lol.